README.md
# Xcat PC control program

https://github.com/skiphansen/xcat_pc

## What is this?

This is a Windows program to control the Xcat Syntor X code plug replacement 
project via a serial port.

This program was originally written in 2004 and hasn't been touched since 
Jan 2009. 

I've put this on github as a history preservation project.

The Sytnor X was OLD in 2004 when this project was started and 18 years later
they are STILL working! In fact I'm listening to 445.42 on my Syntor X base as 
I write this file.

I **WISH** PCs lasted a bit longer, NONE of my PCs from 2004 still work.

## Building

1. Install Visual C++ 6.0 (1998).  Newer versions may be able to convert the
project, I don't know.  I haven't done any Windows programming for years.
2. Clone this repository.
3. Change directory into xcat_pc.
4. Open xcat.dsp.
5. Select "Rebuild All" from the Build menu

## Screen shots








Changelog
####################################
Changes in version 0.25 7/4/2008

New features:
1. Added support for low power transmit Power Control bit used by some T53 and 
   T83 model VHF radios.  

Bug fixes:
1. Corrected error in calculation of code plug data for inverted transmit DPL
   codes.


####################################
Changes in version 0.24 6/15/2008

New features:
1. Added Update Firmware option to the About Tab.  This feature replaces the
   loadhex command line utility.  Unlike the command line utility the PC
   control program restores the Xcat's baudrate and CI-V address to the
   values set prior to the firmware update.  The modes are not saved and
   restored since future firmware updates will no longer clobber them.

Bug fixes:
1. Corrected the "Carrier" selection in Rx DPL and Inverted Rx DPL modes. 
   Previously attempts to select carrier squelch in DPL mode resulted in 
   the Syntor being programmed with an invalid DPL code of "000" rather
   than carrier squelch.
   
2. Changed the "funny bit" settings for DCS codes 251 and 243 to match the
   settings generated by a genuine Motorola suitcase programmer.  Thanks to
   Steve Jordan and Andy Brinkley NC4AB for programming sample code plugs
   to help extract this information.  Due to the tedious and error prone 
   nature of using the suitcase programmer the following codes remain to be 
   verified: 025, 036, 351.

####################################
Changes in version 0.23+ 5/24/2008

Minor enhancements: 
1. Added code to display the loader's firmware version in the about box 
   when available (XCat firmware V 0.30 and later).
2. Modified CAbout::SetActive to always request firmware version info.
3. Enabled Palomar control system to be selected for firmware > 0.30.
4. Added code to configure I/O 7 as a PTT input in Palomar mode.

Bug fixes:
1. Added code to round transmitter offset to OnManualSet.

####################################
Changes in version 0.23 2/2/2008

Bug fixes:
1. Corrected display of out of band receive frequencies.

####################################
Changes in version 0.22 10/6/2007

New features:
1. Added support for digital volume pot in addition to digital squelch pot 
   connected to user function outputs.

####################################
Changes in version 0.20 7/15/2007
1. Added support for digital squelch pot connected to user function outputs.
2. Modified code to allow the PC baudrate to be changed even if communications
   with the Xcat fails.  Previously the PC baudrate was only changed after
   an ACK was received from the Xcat for the new baudrate, this made it
   very difficult to sync baudrates under certain conditions.
3. Added checks for firmware features against Xcat firmware version when
   setting a new configuration.  Configurations that select feature that
   are not supported by a particular firmware release are not longer allowed.

####################################
Changes in version 0.19 2/9/2007

Removed some left over debug code from 0.18 and bumped version number
for release.

####################################
Changes in version 0.18 1/19/2007 (limited beta release)

Bug fixes:
1. Corrected several bugs in the internal code plug data decoding introduced 
   in version 0.17 which caused the transmit offset to be set to "Receive only"
   a frequency was set.
   
2. Corrected rounding errors in calculation of transmit offset frequency from
   code plug read back.
   
Thanks to Tim WD6AWP for reporting the bugs and help testing the fixes.

####################################
Changes in version 0.17 1/2/2007

New features:
1. Added support for DPL (DCS) encoding and decoding to the VFO page.
2. Added a DPL scan option to Scan page. 
3. Added support for UHF range 1 radios (420).  Updated Xcat Firmware required.
4. Added receive only support to the VFO page.  Receive only is now offered
   as a choice in the transmitter offset box.
5. Added support for the transmitter timeout timer to the VFO page.
6. Added code to save retrieved code plug data to c:\xcat.bin when the code
   plug data is read via the Debug Page.
7. Enabled VCO split frequencies to be specified for UHF range radios. Updated 
   Xcat Firmware is required.
8. Corrected the default value of the receive VCO split frequency for VHF range
   2 radios (was 215.9, should be 203.9).

The DPL support is lightly tested.  There is some uncertainty about that correct
value for 3 of the bits that make up the programming information for each code.
Thanks to Andy Brinkley NC4AB DPL codes 023 to 223 match the values generated
by the official Motorola suit case programmer.  The remaining codes may or may 
not work with other than Motorola equipment or Yeasu VX-5R used by Dennis 
KB8ZQZ to empirically determine working values.  The Xcat control program will 
be updated as more information becomes available.

Support for 420 Mhz and UHF VCO split frequencies was added in version 0.23 
of the Xcat firmware which has not been released yet. Please contact me if 
you would like to help test these new features.

####################################
Changes in version 0.16 2/11/2005 (limited beta release)

Minor enhancements: 
1. Added display of new serial frame size debug variable in debug page.

####################################
Changes in version 0.15 1/8/2005

New features:
1. Added the ability to label modes by name.
2. Added the ability to save and restore label name to the configuration
   page.
3. The VFO page now sets the initial dialog values based on code plug
   data read from the Xcat.
4. A recall mode button was added to the VFO page.
5. The ability to set the Xcat's CI-V address and baudrate was added. 
   (Requires Xcat firmware 0.13 or later)

####################################
Changes in version 0.14 12/31/2004 (limited beta release)

Minor enhancements: 
1. Disabled signal reports in debug page.
2. Added display of successfully rx, tx frequency sets to sync data
   debug page.


xcatdlg.h
// $Log: xcatdlg.h,v $
// Revision 1.11  2009/01/06 23:23:42  Skip
// 1. Added support for low power transmit Power Control bit used by some T53 and
//    T83 model VHF radios.
// 2. Corrected error in calculation of code plug data for inverted transmit DPL
//    codes.
//
// Revision 1.10  2008/06/14 14:36:47  Skip
// 1. Added pDownloader member to CAbout class.
// 2. Added OnDownloadChar member to CXcatDlg class.
//
// Revision 1.9  2008/06/01 14:03:09  Skip
// Added UpdateIO5and7, OnUpdateFirmware.
//
// Revision 1.8  2008/02/02 17:58:22  Skip
// Added support for volume pot (not tested).
//
// Revision 1.7  2007/07/15 14:26:21  Skip
// Added squelch pot support.
//
// Revision 1.6  2007/01/02 17:28:56  Skip
// 1. Added support for DPL (DCS) encoding and decoding to the VFO page.
// 2. Added a DPL scan option to Scan page.
// 3. Added support for UHF range 1 radios (420).  Updated Xcat Firmware required.
// 4. Added receive only support to the VFO page.  Receive only is now offered
//    as a choice in the transmitter offset box.
// 5. Added support for the transmitter timeout timer to the VFO page.
// 6. Added code to save retrieved code plug data to c:\xcat.bin when the code
//    plug data is read via the Debug Page.
// 7. Enabled VCO split frequencies to be specified for UHF range radios. Updated
//    Xcat Firmware is required.
// 8. Corrected the default value of the receive VCO split frequency for VHF range
//    2 radios (was 215.9, should be 203.9).
//
// Revision 1.5  2005/01/08 19:31:30  Skip
// 1. Replaced CCommSetup with new dialog that allows XCat's address and
//    baudrate to be configured.
// 2. Replaced recall mode dialog with new class that supports labeling modes.
//
// Revision 1.4  2004/12/27 05:55:24  Skip
// Version 0.13:
// 1. Fixed crash in Debug mode caused by calling ScanPage.ModeData()
//    being called when not on the scan page.
// 2. Added support for sync rx debug data (requires firmware update as well).
// 3. Added request buttons to Debug mode for code plug data and sync rx
//    debug data.
// 4. Corrected bug in configuration of remote base  #4 in Palomar mode.
//
// Revision 1.3  2004/08/28 22:31:31  Skip
// Added the ability to change the serial port baudrate and the address used
// by the Xcat on the bus.
//
// Revision 1.2  2004/08/08 23:41:28  Skip
// Complete implementation of mode scan.
//
// Revision 1.1.1.1  2004/07/09 23:12:30  Skip
// Initial import of Xcat PC control program - V0.09.
// Shipped with first 10 Xcat boards.
//
//
#if !defined(AFX_XCATDIALOG_H__712BDAC8_AF96_4012_A7AA_75BC2D3AC619__INCLUDED_)
#define AFX_XCATDIALOG_H__712BDAC8_AF96_4012_A7AA_75BC2D3AC619__INCLUDED_
#include "loadhex.h"

#if _MSC_VER > 1000
#pragma once
#endif // _MSC_VER > 1000
// XcatDialog.h : header file
//

class CScanEnable;

/////////////////////////////////////////////////////////////////////////////
// CConfigure dialog

class CConfigure : public CPropertyPage
{
   DECLARE_DYNCREATE(CConfigure)

// Construction
public:
   CConfigure();
   ~CConfigure();
   void ConfigMsgRx(unsigned char *Config);
   void VCOSplits(unsigned char *Splits);
   void ModeData(unsigned char *Data);

// Dialog Data
   //{{AFX_DATA(CConfigure)
	enum { IDD = IDD_CONFIGURE };
	CComboBox	mOut7;
	CComboBox	mOut6;
	CComboBox	mOut4;
	CComboBox	mOut3;
   CStatic  mTransferStatus;
   CComboBox   mOut5;
   CComboBox   mOut2;
   CComboBox   mOut1;
   CComboBox   mOut0;
   CComboBox   mControlSys;
   CComboBox   mBand;
   double   mRxVcoSplitFreq;
   BOOL  mSendCosMsg;
   double   mTxVcoSplitFreq;
	BOOL	mUFasSquelch;
	BOOL	mEnableVolumePot;
	//}}AFX_DATA

   int mMode;
   FILE *mFp;
   bool m_bSaving;
   unsigned char mModeData[16];

// Overrides
   // ClassWizard generate virtual function overrides
   //{{AFX_VIRTUAL(CConfigure)
   public:
   virtual BOOL OnSetActive();
   protected:
   virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
   //}}AFX_VIRTUAL

// Implementation
protected:
   // Generated message map functions
   //{{AFX_MSG(CConfigure)
   afx_msg void OnRange1();
   afx_msg void OnRange2();
   afx_msg void OnSelchangeBand();
	afx_msg void OnSelchangeOut3();
	afx_msg void OnSelchangeOut4();
	afx_msg void OnSelchangeOut6();
	afx_msg void OnSelchangeControlSys();
	//}}AFX_MSG
   DECLARE_MESSAGE_MAP()

   void InitButtons();
   void OnSetConfig();
   void OnGetConfig();
   void EnableSplitItems();
   void OnRestoreCodePlug();
   void OnSaveCodePlug();
   void SendModeData();
	int  GetSelectedConfig();
	void SaveSplits();
	void OnSelchangeOut3_4_6(int NewSelection);
	void UpdateUFasSquelch();
	void UpdateEnableVolumePot();
	void UpdateIO5and7();
};

/////////////////////////////////////////////////////////////////////////////
// ManualPage dialog

class ManualPage : public CPropertyPage
{
   DECLARE_DYNCREATE(ManualPage)

// Construction
public:
   ManualPage();
   ~ManualPage();

	void ModeData();
	void UpdatePots();

// Dialog Data
   //{{AFX_DATA(ManualPage)
	enum { IDD = IDD_VFO };
	CSliderCtrl	mVolumePot;
	CSliderCtrl	mSquelchPot;
	CComboBox	mTxTimeout;
   CComboBox   mTxOffset;
   CComboBox   mTxPL;
   CComboBox   mRxPL;
   double   mRxFrequency;
   double   mTxOffsetFreq;
	BOOL	mLowPowerTx;
	//}}AFX_DATA
	bool mbRxDPL;
	bool mbInvRxDPL;
	bool mbTxDPL;
	bool mbInvTxDPL;

	bool mbTxCBMode;
	bool mbLastTxCBMode;
	
	bool mbRxCBMode;
	bool mbLastRxCBMode;

// Overrides
   // ClassWizard generate virtual function overrides
   //{{AFX_VIRTUAL(ManualPage)
   public:
   virtual BOOL OnSetActive();
   protected:
   virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
   //}}AFX_VIRTUAL

// Implementation
protected:
   // Generated message map functions
   //{{AFX_MSG(ManualPage)
   virtual BOOL OnInitDialog();
	afx_msg void OnRxPlEnable();
	afx_msg void OnTxDplEnable();
	afx_msg void OnRxDplEnable();
	afx_msg void OnTxPlEnable();
	afx_msg void OnTxDplEnableInv();
	afx_msg void OnRxDplEnableInv();
	afx_msg void OnSelchangeRxPl();
	afx_msg void OnSelchangeTxPl();
	afx_msg void OnSelchangeTxOffset();
	afx_msg void OnVScroll(UINT nSBCode, UINT nPos, CScrollBar* pScrollBar);
	//}}AFX_MSG

   float mLastEncodePL;
   float mLastDecodePL;
   double mLastTxOffsetFreq;
   bool mForcedSet;
   void OnManualSet();
   void OnSaveMode();
   void OnRecallMode();
	void UpdateRxCB();
	void UpdateTxCB();

   DECLARE_MESSAGE_MAP()
};

/////////////////////////////////////////////////////////////////////////////
// CScanEnable dialog

class CScanEnable : public CPropertyPage
{
   DECLARE_DYNCREATE(CScanEnable)

// Construction
public:
   CScanEnable();
   ~CScanEnable();
	void ModeData();

// Dialog Data
   //{{AFX_DATA(CScanEnable)
	enum { IDD = IDD_SCAN_ENABLE };
	CComboBox	m2ndPriorityMode;
	CComboBox	mPriorityMode;
	BOOL	m_bScanEnabled;
	BOOL	m_bTalkbackEnabled;
	BOOL	m_bFixedScan;
	//}}AFX_DATA


// Overrides
   // ClassWizard generate virtual function overrides
   //{{AFX_VIRTUAL(CScanEnable)
	public:
	virtual BOOL OnSetActive();
	protected:
   virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
	//}}AFX_VIRTUAL

// Implementation
protected:
   // Generated message map functions
   //{{AFX_MSG(CScanEnable)
	virtual BOOL OnInitDialog();
	afx_msg void OnScanEnabled();
	afx_msg void OnFixedScan();
	afx_msg void OnSelchangePriorityChan();
	//}}AFX_MSG
   DECLARE_MESSAGE_MAP()

	void EnableDisableControls();
	void OnRecallMode();
	void OnSaveMode();
	void OnManualSet();

};
/////////////////////////////////////////////////////////////////////////////
// CBandScan dialog

class CBandScan : public CPropertyPage
{
   DECLARE_DYNCREATE(CBandScan)

// Construction
public:
   CBandScan();
   ~CBandScan();

   void OnTimer();
   void CarrierDetectChange(bool bHaveSignal);
   void SaveLockedOutList();

   bool m_bScanActive;
   bool m_bPlScan;
	bool m_bDPlScan;
   bool  m_bHaveSignal;
   double mCurrentFreq;     // In Mhz
   double mLastBottom;
   double mLastTop;

// Dialog Data
   //{{AFX_DATA(CBandScan)
   enum { IDD = IDD_BAND_SCAN };
   CListBox    mLockoutList;
   CComboBox   mStep;
   double   mBandScanTop;
   double   mBandScanBottom;
   CString  mScanFreq;
   double   mHangTime;
   CString  mPLFreqText;
   //}}AFX_DATA


// Overrides
   // ClassWizard generate virtual function overrides
   //{{AFX_VIRTUAL(CBandScan)
   public:
   virtual BOOL OnSetActive();
   virtual BOOL OnKillActive();
   protected:
   virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
   //}}AFX_VIRTUAL

// Implementation
protected:
   void InitButtons();
   // Generated message map functions
   //{{AFX_MSG(CBandScan)
   virtual BOOL OnInitDialog();
   afx_msg void OnSetfocusBandScanTop();
   afx_msg void OnSetfocusBandScanBottom();
   afx_msg void OnSetfocusBandScanStep();
   afx_msg void OnSetfocusLockoutList();
   afx_msg void OnDestroy();
   afx_msg void OnBandScan();
   afx_msg void OnDoPlScan();
	afx_msg void OnDoDplScan();
	//}}AFX_MSG
   void OnStart();
   void OnLockout();
   void ChangeFrequency();
   bool LockedOutFreq();
   void OnDelLockout();
   void ChangePL(bool bSetPl);
	void ChangeDPL();
   void OnSkip();

   DECLARE_MESSAGE_MAP()

   int   mFreqStep;        // In hz
   bool  m_bListSelected;
   int mPlFreq;
   int mDPlFreq;
};

/////////////////////////////////////////////////////////////////////////////
// CCommSetup1 dialog

class CCommSetup1 : public CPropertyPage
{
   DECLARE_DYNCREATE(CCommSetup1)

// Construction
public:
	CCommSetup1();   // standard constructor

// Dialog Data
	//{{AFX_DATA(CCommSetup1)
	enum { IDD = IDD_COMM_SETUP1 };
	CString	mXCatAdr;
	//}}AFX_DATA

	int mNewXcatAdr;
	int mNewBaudrate;
	int mNewComPort;

// Overrides
	// ClassWizard generated virtual function overrides
	//{{AFX_VIRTUAL(CCommSetup1)
	public:
	virtual BOOL OnSetActive();
	protected:
	virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
	//}}AFX_VIRTUAL

// Implementation
protected:
	#define MAX_COM_PORTS	16
	int mPortLookup[MAX_COM_PORTS];

	// Generated message map functions
	//{{AFX_MSG(CCommSetup1)
	virtual BOOL OnInitDialog();
	//}}AFX_MSG
	DECLARE_MESSAGE_MAP()
	void OnSet();
};

/////////////////////////////////////////////////////////////////////////////
// CCommSetup dialog

class CCommSetup : public CPropertyPage
{
   DECLARE_DYNCREATE(CCommSetup)

// Construction
public:
   CCommSetup();
   ~CCommSetup();

// Dialog Data
   BYTE mComPort;

   //{{AFX_DATA(CCommSetup)
	enum { IDD = IDD_COMM_SETUP };
	CString	mXCatAdr;
	//}}AFX_DATA


// Overrides
   // ClassWizard generate virtual function overrides
   //{{AFX_VIRTUAL(CCommSetup)
	public:
   virtual BOOL OnSetActive();
	protected:
   virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
	//}}AFX_VIRTUAL

// Implementation
protected:
   // Generated message map functions
   //{{AFX_MSG(CCommSetup)
      // NOTE: the ClassWizard will add member functions here
   afx_msg void OnCom1();
   afx_msg void OnCom2();
   afx_msg void OnCom3();
   afx_msg void OnCom4();
   afx_msg void OnProperties();
   virtual BOOL OnInitDialog();
   //}}AFX_MSG
   DECLARE_MESSAGE_MAP()

	void OnSet();

   void GrayCommButtons();
   static int ComPortResourceID[];
};

/////////////////////////////////////////////////////////////////////////////
// CAbout dialog

class CAbout : public CPropertyPage
{
   DECLARE_DYNCREATE(CAbout)

// Construction
public:
   CAbout();
   ~CAbout();
	CLoadHex *pDownloader;

// Dialog Data
   //{{AFX_DATA(CAbout)
   enum { IDD = IDD_ABOUTBOX };
   CStatic  mCompiled;
   //}}AFX_DATA

// Overrides
   // ClassWizard generate virtual function overrides
   //{{AFX_VIRTUAL(CAbout)
   public:
   virtual BOOL OnSetActive();
   protected:
   virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
   //}}AFX_VIRTUAL

// Implementation
protected:
   // Generated message map functions
   //{{AFX_MSG(CAbout)
   virtual BOOL OnInitDialog();
	afx_msg void OnUpdateFirmware();
	//}}AFX_MSG
   DECLARE_MESSAGE_MAP()

};

/////////////////////////////////////////////////////////////////////////////
// CDebugMsgs dialog

class CDebugMsgs : public CPropertyPage
{
   DECLARE_DYNCREATE(CDebugMsgs)

// Construction
public:
   CDebugMsgs();
   ~CDebugMsgs();
   void ModeData(unsigned char *Data);
   void SignalReport(int Mode,int bSignal);
	void SyncDebugData(unsigned char *Data);

// Dialog Data
   //{{AFX_DATA(CDebugMsgs)
   enum { IDD = IDD_DEBUG};
   CEdit mEdit;
   //}}AFX_DATA


// Overrides
   // ClassWizard generate virtual function overrides
   //{{AFX_VIRTUAL(CDebugMsgs)
   public:
   virtual BOOL OnSetActive();
   protected:
   virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
   //}}AFX_VIRTUAL

// Implementation
protected:
	void OnGetSyncData();
	void OnGetCodePlugData();
	void OnGetSigReport();

   // Generated message map functions
   //{{AFX_MSG(CDebugMsgs)
      // NOTE: the ClassWizard will add member functions here
   //}}AFX_MSG
   DECLARE_MESSAGE_MAP()

};

// CXcatDlg

class CXcatDlg : public CPropertySheet
{
   DECLARE_DYNAMIC(CXcatDlg)

// Construction
public:
   CXcatDlg(LPCTSTR pszCaption, CWnd* pParentWnd = NULL, UINT iSelectPage = 0);

// Attributes

public:

   ManualPage ManualPage;
   CScanEnable ScanPage;
   CBandScan BandScan;
   CCommSetup1 CommSetup;
   CDebugMsgs DebugMsgs;
   CConfigure Configure;
   CAbout CAbout;

// Operations
public:

// Overrides
   // ClassWizard generated virtual function overrides
   //{{AFX_VIRTUAL(CXcatDlg)
   public:
   virtual BOOL OnInitDialog();
   protected:
   //}}AFX_VIRTUAL

// Implementation
public:
   virtual ~CXcatDlg();

   // Generated message map functions
protected:
   //{{AFX_MSG(CXcatDlg)
   afx_msg void OnPaint();
   afx_msg HCURSOR OnQueryDragIcon();
   afx_msg void OnTimer(UINT nIDEvent);
   //}}AFX_MSG
   DECLARE_MESSAGE_MAP()

   bool bHaveSignal;
   time_t   SignalLostTime;

   LRESULT OnRxMsg(WPARAM wParam,LPARAM lParam);
   LRESULT OnCommError(WPARAM /* wParam*/, LPARAM lParam);
	LRESULT OnDownloadChar(WPARAM /* wParam*/, LPARAM lParam);
};

/////////////////////////////////////////////////////////////////////////////
#endif // !defined(AFX_XCATDIALOG_H__712BDAC8_AF96_4012_A7AA_75BC2D3AC619__INCLUDED_)
/////////////////////////////////////////////////////////////////////////////
// CModeSel dialog

class CModeSel : public CDialog
{
// Construction
public:
	CModeSel(CWnd* pParent = NULL);   // standard constructor

// Dialog Data
	//{{AFX_DATA(CModeSel)
	enum { IDD = IDD_MODE_SEL };
	CComboBox	mMode;
	//}}AFX_DATA


// Overrides
	// ClassWizard generated virtual function overrides
	//{{AFX_VIRTUAL(CModeSel)
	protected:
	virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
	//}}AFX_VIRTUAL

// Implementation
protected:

	// Generated message map functions
	//{{AFX_MSG(CModeSel)
	afx_msg void OnEditName();
	virtual BOOL OnInitDialog();
	virtual void OnOK();
	//}}AFX_MSG
	DECLARE_MESSAGE_MAP()
};
/////////////////////////////////////////////////////////////////////////////
// CEditModeName dialog

class CEditModeName : public CDialog
{
// Construction
public:
	CEditModeName(CWnd* pParent = NULL);   // standard constructor

// Dialog Data
	//{{AFX_DATA(CEditModeName)
	enum { IDD = IDD_NAME_EDIT };
	CString	mName;
	//}}AFX_DATA


// Overrides
	// ClassWizard generated virtual function overrides
	//{{AFX_VIRTUAL(CEditModeName)
	protected:
	virtual void DoDataExchange(CDataExchange* pDX);    // DDX/DDV support
	//}}AFX_VIRTUAL

// Implementation
protected:

	// Generated message map functions
	//{{AFX_MSG(CEditModeName)
	//}}AFX_MSG
	DECLARE_MESSAGE_MAP()
};
/////////////////////////////////////////////////////////////////////////////
// CXcatFileDialog dialog

class CXcatFileDialog : public CFileDialog
{
	DECLARE_DYNAMIC(CXcatFileDialog)

public:
	CXcatFileDialog(BOOL bOpenFileDialog, // TRUE for FileOpen, FALSE for FileSaveAs
		LPCTSTR lpszDefExt = NULL,
		LPCTSTR lpszFileName = NULL,
		DWORD dwFlags = OFN_HIDEREADONLY | OFN_OVERWRITEPROMPT,
		LPCTSTR lpszFilter = NULL,
		CWnd* pParentWnd = NULL);

protected:
	BOOL OnFileNameOK();
	//{{AFX_MSG(CXcatFileDialog)
		// NOTE - the ClassWizard will add and remove member functions here.
	//}}AFX_MSG
	DECLARE_MESSAGE_MAP()
};
